### PR TITLE
Upgrade Proxy and Multisig pallets to Benchmarking V2

### PR DESCRIPTION
**Pull request by re-gius**
Sub-task of PR https://github.com/paritytech/polkadot-sdk/pull/5995

## Integration

These changes should **not** require any integration effort.

### CODE REVIEW

File: substrate/frame/multisig/src/benchmarking.rs [hunk 1]
```diff
@@ -20,8 +20,7 @@
 #![cfg(feature = "runtime-benchmarks")]
 
 use super::*;
-#[allow(deprecated)]
-use frame::benchmarking::v1::*;
+use frame::benchmarking::v2::*;
```
Comment (seadanda) [inline substrate/frame/multisig/src/benchmarking.rs:23]:
```suggestion
use frame_benchmarking::v2::*;
```

File: substrate/frame/proxy/src/benchmarking.rs [hunk 2]
```diff
@@ -22,8 +22,7 @@
 use super::*;
 use crate::Pallet as Proxy;
 use alloc::{boxed::Box, vec};
-#[allow(deprecated)]
-use frame::benchmarking::v1::*;
+use frame::benchmarking::v2::*;
```
Comment (seadanda) [inline substrate/frame/proxy/src/benchmarking.rs:25]:
```suggestion
use frame_benchmarking::v2::*;
```

**Review by seadanda — COMMENTED** [review]
Looks like a few things need to be tidied up - make sure you're passing `--features=runtime-benchmarks` when you check

File: substrate/frame/multisig/src/benchmarking.rs (hunk 1 above)
Comment (seadanda) [reply substrate/frame/multisig/src/benchmarking.rs:23]: Ah wait that's the umbrella crate. You should be able to use that too, `v2` doesn't exist at the top level though, I think you get it via the `prelude`

**Review by gui1117 — APPROVED** [review]
CI needs to pass, but looks good otherwise.

File: substrate/frame/multisig/src/benchmarking.rs (hunk 1 above)
Comment (re-gius) [reply substrate/frame/multisig/src/benchmarking.rs]: fixed in https://github.com/paritytech/polkadot-sdk/pull/6018/commits/57828cd5f8fb60b691c260bd34a0da859469038c

**Review by kianenigma — APPROVED** [review]
Thank you! 

Last question: 

Please check the rust-doc of the each macro respectively, and see if the docs are sufficient and/or can be improved? Some ideas
* Does the v1 one mention the existence of v2? 
* Is there a good "migration" guide available or noted anywhere? 


If CI checks are not relevant, we can merge this anyways, as the target branch is not master.

**Review by shawntabrizi — APPROVED** [review]

**Pull request merged**